src/lib/customclasspath/RegistryJsonHandler.ts

```diff
@@ -0,0 +1,118 @@
+import fs = require('fs');
+import path = require('path');
+import util = require('util');
+
+
+export const CUSTOM_CLASSPATH_REGISTER = path.join('pmd-cataloger', 'catalogs', '.CustomPaths.json');
+export class RegistryJsonHandler {
+
+    private jsonFileOperator: JsonFileOperator;
+
+    constructor() {
+        this.jsonFileOperator = new JsonFileOperator();
+    }
+
+    async readCurrentEntries(): Promise<Map<string, Map<string, string[]>>> {
+        const jsonData = await this.jsonFileOperator.readJsonFile();
+        const mapFromFile = this.jsonToMap(jsonData);
+
+        return mapFromFile;
+    }
+
+    async updateEntries(engineToLanguageMap: Map<string, Map<string, string[]>>) {
+        const jsonObj = this.mapToJson(engineToLanguageMap);
+        await this.jsonFileOperator.writeToJsonFile(jsonObj);   
+    }
+
+    /**
+     * Sample JSON for reference:
+     * {
+     *  "engine1": {
+     *      "language1": ["/path/to/resource1", "path/to/resource2"],
+     *      "language2": ["/path/to/another/resource"]
+     *  },
+     *  "engine2": {
+     *      "language2": ["/yet/another/path"]
+     *  }
+     * }
+     * 
+     * This is converted into a Map such as:
+     * engine ==> {language ==> paths[]}
+     */
+
+    /**
+     * Converts Registry map into JSON object
+     */
+    private mapToJson(engineToLanguageMap: Map<string, Map<string, string[]>>) {
+        let engineLevelJson = {};
+
+        engineToLanguageMap.forEach((languageToPathMap, engine) => {
+            engineLevelJson[engine] = this.getJsonOfLanguageMap(languageToPathMap);
+        });
+
+        return engineLevelJson;
+    }
+
+
+    getJsonOfLanguageMap(languageToPathMap: Map<string, string[]>) {
```

Comment:
I recommend renaming this method, but I'm not sure what its name should be.

Reply:
This method is kind of weird. I'm invoking this to build JSON string for building catalog, so wanted it to have a more identifiable name. I want to revisit this as well in my next iteration. Will add a TODO to track.

Reply:
The convention for methods that turn an object into a string is `toString()`. Which raises an interesting point. What if we replaced all of this stuff about JSONs and Maps with a class called something like `CustomClasspathRegister`, and that was a wrapper around the data, exposing methods like `getLanguagePathsForEngine()`, `addLanguagePathsForEngine()`, and `toString()`? It's a bit late for me currently, but I'm happy to spend some time tomorrow talking through that and maybe mocking something up so you can see what I mean.